other-docs/getting-started/custom-modules.md

```diff
@@ -0,0 +1,165 @@
+---
+order: 30
+---
+# Custom Modules
+
+Platform is designed in a modular way, allowing your project to turn on and off individual modules as appropriate. You can use this same infrastructure for your own custom modules, allowing you to take advantage of frameworks such as the configuration and documentation modules.
+
+
+## Location
+
+Custom modules can be created in one of two ways:
+
+* For project-specific modules, modules should be stored in the project's repository, and loaded in manually.
+* For reusable modules, modules should be created as separate repositories, and published to a Composer repository.
+
+We recommend developing modules following the project-specific guide first, then converting to reusable modules later.
+
+
+## Project-Specific Modules
+
+To start a new project-specific module to your project, first create a directory for the module to live in. We recommend using `content/mu-plugins/{your-module-name}`, to match the commonly-used [WordPress must-use plugins](https://codex.wordpress.org/Must_Use_Plugins) pattern.
+
+Your module's file structure can be structured however you like, but we recommend following the [Human Made plugin structure](https://engineering.hmn.md/standards/structure/#plugin-structure). This structure ensures your code remains maintainable.
+
+There are two basic parts required to creating a module: load your module's file in using the Composer autoloader, and register your module with the Platform core. We recommend having a main entrypoint file (conventionally called `load.php`) which handles loading and registering your module, with other code split into functions and classes in separate files (conventionally within an `inc` directory).
+
+
+### Loading Your Module
+
+In order to use your module, the module needs to be loaded in by the Composer autoloader. This allows the module to register itself with the Platform core, as well as load in any functions or classes it needs.
+
+To load in your module's entrypoint file, add it to a `autoload.files` entry in your project's `composer.json`. For example, for a module called `your-module` with an entrypoint file called `load.php`, your `composer.json` should contain:
+
+```json
+"autoload": {
+    "files": [
+        "content/mu-plugins/your-module/load.php"
+    ]
+}
+```
+
+After adding this, run `composer dump-autoload` to regenerate the autoload files for your project.
+
+
+### Registering Your Module
+
+Within your module's entrypoint file, your module needs to register itself with the Platform core. To do this, you need to add an action on `hm-platform.modules.init` which calls `HM\Platform\Module::register`. This function call registers the module as well as any default configuration and a callback for the module.
+
+A basic registration call looks like:
+
+```php
+use function HM\Platform\register_module;
```

Comment:
Probably add a namespace and opening `<?php` here so it's more copy & pastable

Reply:
I kept this one minimal as it's more for documenting the specifics of the function; the example below is better for copy-and-pasting.